plugins/importexport/doaj/DOAJPlugin.inc.php

```diff
@@ -103,6 +144,326 @@ function exportJournal(&$journal, $outputFile = null) {
 		}
 		return true;
 	}
-}
 
-?>
+	/**
+	 * @param $request PKPRequest
+	 * @param $selectedObjects array
+	 */
+	function _markRegistered($request, $selectedObjects) {
+
+		$articleDao = DAORegistry::getDAO('ArticleDAO');
+		$this->registerDaoHook('ArticleDAO');
+
+		// check for articles
+		$selectedArticles = $selectedObjects[DOAJ_EXPORT_ARTICLES];
+		if (is_array($selectedArticles) && !empty($selectedArticles)) {
+
+			foreach($selectedArticles as $articleId) {
+				$article = $articleDao->getArticle($articleId);
+				$article->setData('doaj::registered', 1);
+				$articleDao->updateArticle($article);
+			}
+		}
+
+		// check for issues
+		$selectedIssues = $selectedObjects[DOAJ_EXPORT_ISSUES];
+		if (is_array($selectedIssues) && !empty($selectedIssues)) {
+			$issueDao = DAORegistry::getDAO('IssueDAO');
+
+			foreach($selectedIssues as $issueId) {
+				$articles = $this->_retrieveArticlesByIssueId($issueId);
+				foreach($articles as $article) {
+					$article->setData('doaj::registered', 1);
+					$articleDao->updateArticle($article);
+				}
+			}
+		}
+
+		// show message & redirect
+		$this->_sendNotification(
+			$request,
+			'plugins.importexport.doaj.markRegistered.success',
+			NOTIFICATION_TYPE_SUCCESS
+		);
+		
+		switch($request->getUserVar('target')) {
```

Comment:
Should include a default case with assert(false). Also, initialize $action before the switch() or else PHP Scrutinizer will complain about a potentially uninitiailized variable being used.